src/test/java/io/cdap/plugin/servicenow/source/ServiceNowMultiRecordReaderTest.java

```diff
@@ -198,6 +198,6 @@ public void testFetchDataOnInvalidTable() throws IOException, OAuthProblemExcept
     } catch (RuntimeException | OAuthProblemException | OAuthSystemException e) {
       Assert.assertTrue(e instanceof RuntimeException);
     }
-    Assert.assertFalse(serviceNowMultiRecordReader.nextKeyValue());
```

Comment:
I didn't understand this part change. Can you explain?

Reply:
Ideally we should not be calling `recordReader.nextValue()` if `recordReader.initialize()` fails. And it should fail if the the function is called.